player.go

```diff
@@ -15,11 +15,64 @@ import (
 // playerSpeed is the distance the player moves per update cycle
 const playerSpeed float64 = 2
 
+//
+const (
+	playerIdle    int = 0
+	playerWalking     = 1   
+)
+
 // Player is the player character in the game
 type Player struct {
-	Object *resolv.Object
-	Angle  float64
-	Sprite *SpriteSheet
+	Object        *resolv.Object
+	Angle         float64
+	Frame         int
+	State         int
+	Sprite        *SpriteSheet
+}
+
+// Update updates the state of the player
+func (p *Player) Update(g *Game) {
+	state := playerIdle
+
+	if ebiten.IsKeyPressed(ebiten.KeyW) {
+		p.MoveUp()
+		state = playerWalking
```

Comment:
If you put this state change inside `player.move()` then you wouldn't have to write it out all 4 times, only once.

Reply:
Yep, I wasn't happy with it either. OK, I move it to move() for the time being